src/statemachine.rs

```diff
@@ -213,16 +236,21 @@ impl StateMachine {
 
         // Check if the combination of device-protection and request-options
         // are allowing for 'discouraged', meaning no auth required.
-        if cmd.can_skip_user_verification(info) {
+        if cmd.can_skip_user_verification(info, uv_req) {
             return Ok(PinUvAuthResult::NoAuthRequired);
         }
 
         // Device does not support any auth-method
-        if !pin_configured && !supports_uv {
-            // We'll send it to the device anyways, and let it error out (or magically work)
+        if (skip_uv || !supports_uv) && !supports_pin {
```

Comment:
@msirringhaus Can you think about whether we need to call `cmd.set_uv_option(Some(true))` here in the uv required case? And if so, what if `!supports_uv`. Do we need to error-out then?

Reply:
Hm, tricky. 

I'd say, if `supports_uv=true`, set the option. The device should then error out with `UvBlocked` or the like. As for PINs, Firefox should have popups to inform the user about that.

If `supports_uv=false`, we have two options:
* If we do NOT set it, then there is a good chance, the request succeeds, but the response will have `uv=false`. Some servers may still accept that (wrongly), and would mean less friction for the user. If the server does not accept it, it will error out (with a server-side error message). 
* If we DO set it, the device will very likely error out with `InvalidOption` or the like.

In principle, I'd favor setting it, and let  the device error out how it sees fit. The problem may here be: how do we inform the user about it in a sensible way? But that is a 'Firefox-side problem' so to speak. I think for the library, it should probably let the device error out.

Reply:
I agree with setting it. Looking a little closer at the spec ([makeCred 11.2](https://fidoalliance.org/specs/fido-v2.1-ps-20210615/fido-client-to-authenticator-protocol-v2.1-ps-errata-20220621.html#op-makecred-step-protected) and [getAssertion 6.2](https://fidoalliance.org/specs/fido-v2.1-ps-20210615/fido-client-to-authenticator-protocol-v2.1-ps-errata-20220621.html#sctn-getAssert-authnr-alg)), a conformant authenticator that supports uv should return `CTAP2_ERR_PIN_BLOCKED` if the uv retry counter is 0, which is the likely case when `supports_uv = true` and `skip_uv`. We show a device blocked notificaiton in Firefox when that happens.